authors/apps/articles/views.py

```diff
@@ -390,3 +393,71 @@ def retrieve(self, request, slug, *args, **kwargs):
         read = len(words) / 279
         return Response({'read_time': round(read)}, status.HTTP_200_OK)
 
+
+class HighlightText(ListCreateAPIView):
+    permission_classes = (IsAuthenticatedOrReadOnly, )
+    serializer_class = ArticleSerializer
+
+    def create(self, request, slug):
+        """Get an article to comment"""
+        try:
+            # get an article that matches the specified slug
+            # and return a message if the article does not exist
+            single_article_instance = Article.objects.get(slug=slug)
+        except ObjectDoesNotExist:
+            raise NotFound("An article with this slug does not exist")
+
+        # check for all the required fields
+        required_fields = ['start_highlighting', 'end_highlighting', 'body']
+        for field in required_fields:
+            if field not in request.data:
+                return Response({'error': '{} is required'.format(field)},
+                                status.HTTP_400_BAD_REQUEST)
+
+        # serializing article details in order to extract
+        # the body of an article
+        serializer = self.serializer_class(single_article_instance)
+        word = serializer.data['body'].split()
```

Comment:
I don't expect you to split the words though... because a highlight can be from the middle of the word

Reply:
Okay